go/app/infra.go

```diff
@@ -26,76 +34,110 @@ type Item struct {
 type ItemRepository interface {
 	Insert(ctx context.Context, item *Item) error
 	GetAll(ctx context.Context) ([]Item, error)
+	GetByID(ctx context.Context, itemID int) (*Item, error)
+	GetCategoryID(ctx context.Context, categoryName string, categoryID *int) error
```

Comment:
[imo]
新たに取得する変数はポインタで渡すのではなく、他のメソッドのように返り値として返却する方針で良いと思います

```suggestion
	GetCategoryIDByName(ctx context.Context, categoryName string) (int, error)
```